modin/pandas/series.py

```diff
@@ -489,11 +489,14 @@ def apply(self, func, convert_dtype=True, args=(), **kwds):
         # a list or a dictionary, which means that the return type won't change from
         # type(self), so we catch that error and use `self.__name__` for the return
         # type.
-        # Because a `Series` cannot be empty in pandas, we create a "dummy" `Series` to
-        # do the error checking and determining the return type.
+        # We create a "dummy" `Series` to do the error checking and determining
+        # the return type.
         try:
+            # Handling of empty Series
+            test_data = type(self[0])() if len(self.index) > 0 else []
```

Comment:
This will take up to 100ms. Is there no way to just create `pandas.Series([], dtype=self.dtype)`? Does the Series require data to work properly?

Reply:
Yes, you are right, this step takes up a lot of time.
I`m afraid yes, we need data for proper work, it can be seen in the example below
```
>>> pandas.Series([], dtype="int64").apply("count")
Series([], dtype: int64)
>>> pandas.Series(0, dtype="int64").apply("count")
1
```
We can speed-up return_type getting by probing numeric and non-numeric dummy Series successively as it is done in the new commit.